src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -569,11 +568,24 @@ private static DetailAST getIdentLastToken(DetailAST ast) {
      * @param leftBracket the ast we are checking
      * @return dot preceding the left bracket
      */
-    private static DetailAST getPrecedingDot(DetailAST leftBracket) {
-        final DetailAST referencedClassDot =
-                leftBracket.getParent().findFirstToken(TokenTypes.DOT);
+    private static Optional<DetailAST> getPrecedingDot(DetailAST leftBracket) {
         final DetailAST referencedMemberDot = leftBracket.findFirstToken(TokenTypes.DOT);
-        return Optional.ofNullable(referencedMemberDot).orElse(referencedClassDot);
+        final Optional<DetailAST> result = Optional.ofNullable(referencedMemberDot);
+        return result.or(() -> getReferencedClassDot(leftBracket));
```

Comment:
Can we inline return value?

Reply:
Also, can we use `orElseGet` to avoid unnecessary method call?

Reply:
I am not sure it will look better.
Method returns Optional, so it really looks ugly when you try to combine it from 2 nullable variables. Therefore I decided it is better to make a separate method.

Reply:
Ok, that makes sense.